app/src/main/java/com/duckduckgo/app/browser/serviceworker/ServiceWorkerLifecycleObserver.kt

```diff
@@ -23,18 +23,15 @@ import androidx.webkit.WebViewFeature
 import androidx.webkit.WebViewFeature.SERVICE_WORKER_BASIC_USAGE
 import com.duckduckgo.app.browser.RequestInterceptor
 import com.duckduckgo.app.global.exception.UncaughtExceptionRepository
-import com.duckduckgo.di.scopes.AppScope
 import timber.log.Timber
 import javax.inject.Inject
-import dagger.SingleInstanceIn
 
-@SingleInstanceIn(AppScope::class)
 class ServiceWorkerLifecycleObserver @Inject constructor(
     private val requestInterceptor: RequestInterceptor,
     private val uncaughtExceptionRepository: UncaughtExceptionRepository,
 ) : DefaultLifecycleObserver {
-    override fun onStart(owner: LifecycleOwner) {
-        super.onStart(owner)
+    override fun onCreate(owner: LifecycleOwner) {
+        super.onCreate(owner)
```

Comment:
Instead of changing the lifecycle, would it be possible to make the `serviceWorkerLifecycleObserver` "lazy" so that `getInstance()` is only called when that class is accessed for the first time?

Sort of like: https://github.com/duckduckgo/Android/pull/1844.

Reply:
Discussed in MM. It wouldn't really make sense to lazily instantiate `serviceWorkerLifecycleObserver` itself since it doesn't hold any internal state and the creation of `ServiceWorkerControllerCompat` (crash point) happens on lifecycle `onCreate() `anyway.